graphql.go

```diff
@@ -178,8 +178,10 @@ func (c *Client) request(ctx context.Context, query string, variables map[string
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		body, _ := io.ReadAll(resp.Body)
-		err := newError(ErrRequestError, fmt.Errorf("%v; body: %q", resp.Status, body))
+		err := newError(ErrRequestError, NetworkError{
+			message:    "non-200 OK status code",
```

Comment:
```suggestion
			message:    string(body),
```

Reply:
Would it be sensible to store the body separately to the message? Large responses (e.g. HTML responses) can become quite unwieldy when using structured logging, for example. This was part of the reason I initially suggested storing a reader, but storing as a string could work too.

e.g. something like this

```go
NetworkError{
	message string
	status int
	body string
}

func (e NetworkError) Body() string {
	return e.body
}
```

Happy to go with your preference if you don't like this suggestion, though.

Reply:
That's a valid point. Btw, we don't need the `message` field with the static `non-200 OK status code` text. We have enough context in the `Error` method 

```go
func (e NetworkError) Error() string {
	return fmt.Sprintf("%d %s", e.statusCode, http.StatusText(e.statusCode))
}
```

Let's remove the `message` field and add a getter method for the body content

Reply:
Great, that works for me too